.github/workflows/test.yaml

```diff
@@ -95,3 +95,67 @@ jobs:
       - uses: ./.github/actions/setup-env
       - name: Run optimized tests
         run: tox -e optimized
+
+  mutation-test:
+    runs-on: ubuntu-latest
+    needs: static
+    if: github.event_name == 'pull_request'
+    strategy:
+      matrix:
+        fork: [frontier, homestead, tangerine_whistle, spurious_dragon, byzantium, constantinople, istanbul, berlin, london, paris, shanghai, cancun, prague, osaka]
```

Comment:
If you generate the matrix from the output of a previous job (i.e. `${{ needs.changed-forks.outputs.changed-forks }}` where 'changed-forks` is a preceding job that stores the changed forks), you can cut down own the number of runs plus use `ethereum_spec_tools.forks:Hardfork` to dynamically generate the list instead of having to hard code it.